next/13/replace-get-search-params/index.ts

```diff
@@ -0,0 +1,312 @@
+import {
+	CallExpression,
+	Collection,
+	File,
+	ImportDeclaration,
+	JSCodeshift,
+} from 'jscodeshift';
+import jscodeshift from 'jscodeshift';
+
+import type { HandleFile, Repomod } from '@intuita-inc/repomod-engine-api';
+
+type Settings = {
+	useCompatSearchParamsHookAbsolutePath: string;
+	useCompatSearchParamsHookModuleSpecifier: string;
+};
+
+type ModFunction<T, D extends 'read' | 'write'> = (
+	j: JSCodeshift,
+	root: Collection<T>,
+	settings: Settings,
+) => [D extends 'write' ? boolean : false, ReadonlyArray<LazyModFunction>];
+
+type LazyModFunction = [
+	ModFunction<any, 'read' | 'write'>,
+	Collection<any>,
+	Settings,
+];
+
+type Dependencies = {
+	jscodeshift: typeof jscodeshift;
+};
+
+type State = {
+	useCompatSearchParamsHookCreated: boolean;
+};
+
+type FileCommand = Awaited<ReturnType<HandleFile<Dependencies, State>>>[number];
+
+export const USE_COMPAT_SEARCH_PARAMS_HOOK_CONTENT = `
+import { ReadonlyURLSearchParams, useParams, useSearchParams } from "next/navigation"
+
+type ParsedUrlQuery = Record<string, string | string[]>;
+
+// ref: https://github.com/vercel/next.js/blob/ab7b0f59fb0d793fb4ee593ae93f7e08bab23b5b/packages/next/src/shared/lib/router/utils/querystring.ts#L3C17-L3C39
+
+export const searchParamsToUrlQuery = (
+    searchParams: ReadonlyURLSearchParams
+  ): ParsedUrlQuery => {
+    const query: ParsedUrlQuery = {}
+
+    searchParams.forEach((value, key) => {
```

Comment:
I think we need to add the licensing information from Vercel.
on the other hand, can we import `searchParamsToUrlQuery`?

Reply:
no they dont export it.